### PR TITLE
feat: expose all parent links

### LINKED ISSUE #35: Expose all incoming child links for QueryPlan nodes
## Background

`QueryPlan.New` currently stores a single `parentMap` entry per child index. Real Spanner plans can contain multiple incoming child links for the same child node, for example a `Serialize Result` linking to the same `Array Subquery` child through both a regular link and a `Scalar` link.

Single-parent lookup is usually enough for tree rendering, but linting, compact summaries, and debugging benefit from seeing every incoming link.

## Proposal

Add a non-breaking API that returns all parent links for a child node, with both the parent node and the child link available.

Possible shape:

```go
type ResolvedParentLink struct {
    Parent    *sppb.PlanNode
    ChildLink *sppb.PlanNode_ChildLink
}

func (qp *QueryPlan) ParentLinks(childIndex int32) []*ResolvedParentLink
```

`IncomingLinks` could be an alias if that reads better for callers.

## Acceptance criteria

- Multiple incoming links to the same child index are preserved and returned in stable plan traversal order.
- Existing parent lookup behavior remains compatible.
- Focused tests cover duplicate incoming links from the same parent and/or multiple parents.
- `go test ./...` passes.

### PR DESCRIPTION
**Pull request by apstndb**
Closes #35

Summary:
- Add `ResolvedParentLink` and `QueryPlan.ParentLinks(childIndex)` to expose every incoming child link for a node.
- Preserve existing `GetParentNodeByChildIndex` behavior by keeping the single-parent map unchanged.
- Preserve stable plan traversal order and return a copied slice to callers.
- Add tests for duplicate incoming links from one parent, links from multiple parents, empty/out-of-range lookups, and single-parent compatibility.

Validation:
- `go test -v ./...`
- `git diff --check`
- OpenCode diff review with `opencode-go/qwen3.6-plus`: no material issues; minor doc/edge-test suggestions addressed

Note:
- `golangci-lint run --timeout=5m` could not be used locally: golangci-lint 2.12.1 reports no Go files to analyze in this checkout.

### CODE REVIEW

File: queryplan.go [hunk 1]
```diff
@@ -12,8 +12,9 @@ import (
 )
 
 type QueryPlan struct {
-	planNodes []*sppb.PlanNode
-	parentMap map[int32]int32
+	planNodes      []*sppb.PlanNode
+	parentMap      map[int32]int32
+	parentLinksMap map[int32][]*ResolvedParentLink
```
[the remark this answers is not in the document]
Comment (apstndb) [reply queryplan.go]: Addressed in f17be36: ParentLinks now stores and returns ResolvedParentLink values, so callers cannot mutate internal link wrapper state through the returned slice.

File: queryplan.go [hunk 2]
```diff
@@ -128,6 +138,21 @@ func (qp *QueryPlan) GetParentNodeByChildLink(link *sppb.PlanNode_ChildLink) *sp
 	return qp.GetParentNodeByChildIndex(link.GetChildIndex())
 }
 
+// ParentLinks returns all parent child links that point to childIndex.
+//
+// Links are returned in plan node traversal order, preserving each parent's
+// ChildLinks order. The returned slice is a copy and can be modified by callers.
+func (qp *QueryPlan) ParentLinks(childIndex int32) []*ResolvedParentLink {
```
[the remark this answers is not in the document]
Comment (apstndb) [reply queryplan.go]: Addressed in f17be36: ParentLinks now returns []ResolvedParentLink and still clones the slice before returning it.

File: queryplan.go [hunk 3]
```diff
@@ -54,10 +56,18 @@ func New(planNodes []*sppb.PlanNode) (*QueryPlan, error) {
 				return nil, fmt.Errorf("%w: parent node %d childLinks[%d] has childIndex %d, len(planNodes)=%d", ErrChildLinkIndexOutOfRange, planNode.GetIndex(), j, childIndex, len(planNodes))
 			}
 			parentMap[childIndex] = planNode.GetIndex()
+			parentLinksMap[childIndex] = append(parentLinksMap[childIndex], &ResolvedParentLink{
+				Parent:    planNode,
+				ChildLink: childLink,
+			})
```
[the remark this answers is not in the document]
Comment (apstndb) [reply queryplan.go]: Addressed in f17be36: New now appends ResolvedParentLink values instead of pointers.

File: queryplan_test.go [hunk 4]
```diff
@@ -117,6 +117,66 @@ func TestHasStats(t *testing.T) {
...
+	}
+
+	if got := qp.GetParentNodeByChildIndex(2).GetIndex(); got != 1 {
+		t.Fatalf("GetParentNodeByChildIndex(2) = %d, want 1", got)
+	}
+
+	got[0] = nil
+	if got := qp.ParentLinks(2)[0]; got == nil {
+		t.Fatal("ParentLinks(2) returned internal slice")
+	}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply queryplan_test.go]: Addressed in f17be36: the slice isolation test now mutates a returned ResolvedParentLink value and verifies subsequent calls are unchanged.
